### ISSUE TITLE
Include "Ideas", "Implementations" and "CFC" stages in Auto-WCAG process

### ISSUE THREAD

**Issue by annethyme** [act-rules/act-rules.github.io#368]
### Background

Several people have lately raised concerns about quality issues in published rules + we have quite a lot of overhead when feedback for rules comes too late, e.g. when an almost complete rule suddenly gives rise to a discussion on whether the rule should have been written at all, or when issues are found in already published rules. 

Building on some of the discussions at the Auto-WCAG face-2-face in late October, Siteimprove would like to suggest these process/workflow additions: 

### Overview of suggested process/workflow
- Ideas looking for initial approvals (NEW)
- To Do
- In Progress
- Needs Reviews
- Ready for Implementation (NEW)
- CFC (NEW)
- Done 
- Rejected

(see this suggested workflow on the board here: https://github.com/auto-wcag/auto-wcag/projects/5)

### Ideas looking for initial approvals
To avoid that people spend a huge amount of time on writing rules and reviewing details in them to then see the whole concept of the rule rejected when looking for a final review, we suggest adding an "Ideas looking for initial approvals" stage. This should give an indication on whether the rule can get 3 final approving reviews and get through CFC later, or if the idea needs to be tweaked or rejected all together.

We suggest creating the pull request for the rule already at this stage, but for content of the rule it should be enough to have e.g. a title, a description and success criteria listed, for reviewers to be able to evaluate the idea for the rule.

### Ready for implementation (OUTDATED)
At the moment, we wait until rules are published before we start implementing them, because we never know when rules are stable enough to start implementing, or if there is still the possibility that they could change radically due to feedback in reviews. 

However, we also often find things that needs to be changed in rules as soon as we start implementing them, e.g. issues with test cases, ambiguities in applicability or expectations, missing definitions etc. 

Right now this means that we have a lot of open issues for already published rules. Making new pull requests for all of these issues and seeking 3 approvals for each change, no matter how minor it is, is a daunting task, since people will start noticing other things too when reviewing, which will either make the pull request expand into other areas, or result in a bunch of new issues, which means that the issues are piling up, and other known issues are not even reported in, because it will just add to the pile of unresolved issues. 

Therefore we suggest moving the implementations phase to BEFORE publishing the rule, so that all the feedback from the implementation can get included in the original pull request.

### CFC
As an addition to getting the 3 approvals (from different organisations), we suggest a CFC stage. 
When a rule has 3 approvals AND some initial implementations, we suggest an email is sent out to all of Auto-WCAG (this could be automated later), giving everyone a chance to review and approve/disapprove the *almost-finished rule*. It is important to us that the broad review is done on the rule in its close to final form. We suggest a 2-4 week grace period for CFC before moving on to next stage.

We also suggest that this CFC step is disconnected from the Auto-WCAG calls, where the agenda should be reserved for items that require discussions and "meta" subjects. 
CFCs require a thorough review, and should therefore not have to fight for space on the agenda with all kinds of other items.
Alternatively there could be a single agenda items that are something like "Questions for open CFCs".

The CFC stage could be both before or after implementations, but having it after would allow the rule to be as stable as possible before going into CFC and allow for people to also have a look at the implementations before making their decision on the rule.

**Comment by annethyme** [comment]
Notes from Auto-WCAG: Skipping Implementations phase, but keeping Ideas phase and CFC phase.
Anne will update proposal.

**Comment by EmmaJP** [comment]
I like the idea of trial implementation feeding into the process, but not if it mean pull requests become stale in the process.

**Comment by annethyme** [comment]
Updated proposal: 

This proposal seeks to address: 
- concerns of quality within published rules 
- overhead for rules writers and implementers when feedback for rules comes too late, e.g. when an almost complete rule suddenly gives rise to a discussion on whether the rule should have been written at all, or when issues are found in already published rules.

### Overview of suggested process/workflow
* Ideas looking for initial approvals (NEW)
* To Do
* In Progress
* Needs Reviews
* CFC (NEW)
* Done
* Rejected

(see this suggested workflow on the board here: https://github.com/auto-wcag/auto-wcag/projects/5)

### New stage: Ideas looking for initial approvals
To avoid that people spend a huge amount of time on writing rules and reviewing details in them to then see the whole concept of the rule rejected when looking for a final review, we suggest adding an "Ideas looking for initial approvals" stage. This should give an indication on whether the rule can get 3 final approving reviews and get through CFC later, or if the idea needs to be tweaked or rejected all together.

**Note:** We suggest creating the pull request (PR) for the rule already at this stage, but for content of the rule it should be enough to have e.g. a title, a description and success criteria listed, for reviewers to be able to evaluate the idea for the rule.
Creating the rule as a pull request already here allows for keeping the discussion of the rule in one place. However, if a rule writer prefers using issues at this stage, this should also be allowed.

### CFC phase
As an addition to getting the 3 approvals (from different organisations), we suggest a CFC (call for consensus) stage.

When a rule has 3 approvals, we suggest an email is sent out to all of Auto-WCAG community group (this could be automated later), giving everyone a chance to review and approve/disapprove the _almost-finished rule_. 
It is important to us that the broad review is done on the rule in its close to final form. 

We suggest a 2-4 week grace period for CFC before moving on to next stage, allowing for: 

- Thorough reviews, giving people time to dive into the rule
- First implementations of rule in tools and methodologies.

We suggest that this CFC step is disconnected from the Auto-WCAG calls, but having an agenda item along the lines of "Questions for open CFCs".

In the CFC stage, we encourage tool vendors and testing methodology owners to implement the rule. 
At this stage, the rule should be quite stable, minimising the risk of doing a too-early implementation, that has to be re-done from scratch later due to changes in the rule.

Implementations are important at this stage, since we often find things that needs to be changed in rules as soon as we start implementing them, e.g. issues with test cases, ambiguities in applicability or expectations, missing definitions etc.

Therefore we encourage initial implementations BEFORE publishing the rule, so that all the feedback from the implementation can get included in the original pull request, instead of being raised as new issues for already published rules - making even published rules quite unstable.

**Open question:** How big changes can we do to a rule as a result of the CFC stage feedback, before we have to send it out in another round of CFC before publishing?

**Comment by annethyme** [comment]
Decided on Auto-WCAG call that rule authors send out CFC email themselves to the mailing list.
CFC is 2 weeks - unless someone asks for extension.
@Brynanders will try it out for https://github.com/auto-wcag/auto-wcag/pull/263 that already has 3 approvals.

**Comment by annethyme** [comment]
@WilcoFiers, https://github.com/auto-wcag/auto-wcag/pull/263 got through the two-week CFC stage. 
Does that mean we can merge it?

**Comment by annethyme** [comment]
The new workflow is already carried out in the CG, and now documented on https://act-rules.github.io/pages/design/process/, so I will close this issue.
